uiautomationutilities/src/main/java/com/microsoft/identity/client/ui/automation/broker/BrokerCompanyPortal.java

```diff
@@ -198,42 +201,15 @@ public void enrollDevice(@NonNull final String username,
 
         handleFirstRun(); // handle CP first run
 
-        // click Sign In button on CP welcome page
-        UiAutomatorUtils.handleButtonClick("com.microsoft.windowsintune.companyportal:id/sign_in_button");
-
-        if (isFederated) {
-            final MicrosoftStsPromptHandlerParameters promptHandlerParameters = MicrosoftStsPromptHandlerParameters.builder()
-                    .prompt(PromptParameter.LOGIN)
-                    .consentPageExpected(false)
-                    .expectingLoginPageAccountPicker(false)
-                    .sessionExpected(false)
-                    .isFederated(true)
-                    .loginHint(null)
-                    .build();
-
-            final MicrosoftStsPromptHandler microsoftStsPromptHandler = new MicrosoftStsPromptHandler(promptHandlerParameters);
-            ((AdfsLoginComponentHandler) microsoftStsPromptHandler.getLoginComponentHandler()).handleEnrollmentPrompt(username, password);
-        } else {
-            final MicrosoftStsPromptHandlerParameters promptHandlerParameters = MicrosoftStsPromptHandlerParameters.builder()
-                    .prompt(PromptParameter.LOGIN)
-                    .consentPageExpected(false)
-                    .expectingLoginPageAccountPicker(false)
-                    .sessionExpected(false)
-                    .loginHint(null)
-                    .build();
-
-            final MicrosoftStsPromptHandler microsoftStsPromptHandler = new MicrosoftStsPromptHandler(promptHandlerParameters);
-
-            Logger.i(TAG, "Handle prompt in AAD login page for enrolling device..");
-            // handle AAD login page
-            microsoftStsPromptHandler.handlePrompt(username, password);
-        }
+        // Company portal password page is somewhat inconsistent, found out turning off battery optimization helps from testing
```

Comment:
interesting

Reply:
CP password page ui is very hit or miss. Not sure why, but we always seem to successfully catch username page, but password page elements are not visible sometimes (quite often honestly). Checking them in uiautomatorviewer when this happens, they don't show up in the node tree view. Battery optimization being off seems to make the problem less common.